datalab_cohorts/expectation_generators.py

```diff
@@ -40,26 +41,29 @@ def generate_dates(population, earliest_date, latest_date, rate):
     (increasingly common)
 
     """
-    assert (
-        rate == "exponential_increase"
-    ), "Only exponential increase currently supported"
     low = datetime.strptime(earliest_date, "%Y-%m-%d").date()
     high = datetime.strptime(latest_date, "%Y-%m-%d").date()
     elapsed_days = (high - low).days
 
-    # We oversample the distribution to trim the long tail of the
-    # exponential function
-    oversample_ratio = 1.5
-    distribution = (
-        expon.rvs(loc=0, scale=0.1, size=int(population * oversample_ratio))
-        * elapsed_days
-    ).astype("int")
-    distribution = distribution[distribution <= elapsed_days]
+    if rate == "exponential_increase":
+        # We oversample the distribution to trim the long tail of the
+        # exponential function
+        oversample_ratio = 1.5
+        distribution = (
+            expon.rvs(loc=0, scale=0.1, size=int(population * oversample_ratio))
+            * elapsed_days
+        ).astype("int")
+        distribution = distribution[distribution <= elapsed_days]
+    elif rate == "uniform":
+        distribution = uniform.rvs(size=int(population)) * elapsed_days
+        distribution = distribution.astype("int")
+    else:
+        print("Only exponential and uniform distributions currently supported")
 
     # And then sample it back down to the requested population size
     distribution = np.random.choice(distribution, population, replace=False)
 
-    df = pd.DataFrame(sorted(distribution), columns=["days"])
+    df = pd.DataFrame(distribution, columns=["days"])
```

Comment:
Why remove the sort?

Reply:
It meant that the dates weren't randomly distributed, meaning that patients at the top of the [csv](https://gist.github.com/alexwalkercebm/8de0c7ec7c89c3184aae5f8d9ddff5bb) all had very recent outcomes and covariates, while ones lower down had ones from further back. Do we not want them all to be randomly distributed?

Reply:
Hm, now you put it like that.... I guess so!